### PR TITLE
add token supply query

### PR DESCRIPTION
**Pull request by 0xArbi**
Pretty much self explanatory, I need to be able to query for the supply of a native token during testing.

I don't know how to get around the `non_exhaustive` issue, that's why I've redefined the `SupplyResponse` struct.

### CODE REVIEW

**Comment by chipshort** [comment]
About that non-exhaustive issue: In newer versions of `cosmwasm-std`, `SupplyResponse` implements `Default`, so you can use that to get one and adjust the `amount` on that:
```rust
let mut res = SupplyResponse::default();
res.amount = amount;
```

**Comment by NoahSaso** [comment]
What is left to fix here? I'm blocked on this, happy to finish up the PR.

File: src/bank.rs [hunk 1]
```diff
@@ -61,12 +73,28 @@ impl BankKeeper {
...
     fn get_balance(&self, bank_storage: &dyn Storage, account: &Addr) -> AnyResult<Vec<Coin>> {
         let val = BALANCES.may_load(bank_storage, account)?;
         Ok(val.unwrap_or_default().into_vec())
     }
 
+    fn get_supply(&self, bank_storage: &dyn Storage, denom: String) -> AnyResult<Coin> {
+        let supply: Uint128 = BALANCES
+            .range(bank_storage, None, None, Order::Ascending)
+            .into_iter()
+            .map(|a| a.unwrap().1)
```
Comment (chipshort) [inline src/bank.rs:85]: I'm not a big fan of this unwrap, the error should be bubbled up

**Review by chipshort — COMMENTED** [review]
Left some comments here to improve.
Also would like to see some tests

**Pull request closed without merging**